### PR TITLE
Linux: add XDG metadata files

### PR DESCRIPTION
**Pull request by eszlari**
- [x] Branch rebased on latest master `git pull --rebase upstream master`
- [ ] Unit tests confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [x] Commit message is [well-written](https://chris.beams.io/posts/git-commit/)

When this is merged, @PrzemoF could remove the files under: https://github.com/FreeCAD/FreeCAD/tree/master/package/fedora

### CODE REVIEW

File: src/XDGData/org.freecadweb.FreeCAD.desktop [hunk 1]
```diff
@@ -0,0 +1,16 @@
+[Desktop Entry]
+Name=FreeCAD
+Name[de]=FreeCAD
+Name[pl]=FreeCAD
```
Comment (luzpaz) [inline src/XDGData/org.freecadweb.FreeCAD.desktop]: should we add other translation languages besides these 2 ?

File: src/XDGData/org.freecadweb.FreeCAD.desktop [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+Comment=Feature based Parametric Modeler
+Comment[de]=Feature-basierter parametrischer Modellierer
+GenericName=CAD Application
+GenericName[de]=CAD-Anwendung
+GenericName[pl]=Aplikacja CAD
```
Comment (luzpaz) [inline src/XDGData/org.freecadweb.FreeCAD.desktop]: should we add other translation languages besides these?

**Comment by luzpaz** [comment]
@eszlari does this PR address https://www.freecadweb.org/tracker/view.php?id=2956 ?

**Comment by eszlari** [comment]
> does this PR address https://www.freecadweb.org/tracker/view.php?id=2956 ?

This flatpak build uses a shell script with `--user-cfg` and `--system-cfg` as a workaround to set the paths correctly. What is still missing is:

```
~/.FreeCAD/Macro
~/.FreeCAD/Mod
~/.FreeCAD/cookies
```

FreeCAD should use `QStandardPaths` for all these files/folders:
http://doc.qt.io/qt-5/qstandardpaths.html

**Comment by yorikvanhavre** [comment]
> ~/.FreeCAD/Macro
> ~/.FreeCAD/Mod
> ~/.FreeCAD/cookies

These are not part of the installation... They are user files created at runtime. Different issue, not related with this

**Comment by eszlari** [comment]
Yeah, you are right, I'm just working on the flatpak at the moment, and faced this issue too.

Thanks for the merge! Could you update the `appdata.xml` file when the next stable release comes out?

**Comment by eszlari** [comment]
If you or any other FreeCAD developer would like to have commit access to the FreeCAD Flatpak repository on Flathub, please add a comment on this PR:

https://github.com/flathub/flathub/pull/683

You can test the flatpak locally by:

```
$ git clone https://github.com/eszlari/flathub
$ cd flathub
$ git checkout freecad
$ git submodule update
$ flatpak remote-add --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo
$ flatpak install --user flathub org.kde.Sdk//5.11
$ flatpak-builder --user --ccache --install --force-clean build_dir org.freecadweb.FreeCAD.yaml
$ flatpak run org.freecadweb.FreeCAD
```

**Comment by luzpaz** [comment]
> Could you update the appdata.xml file when the next stable release comes out?

@yorikvanhavre FYI, we should add this to a Release TODO list so we don't forget during release time, what do you think ?

**Comment by PrzemoF** [comment]
I'm a bit confused with the review system. Anyway, there is a PR that implements using the new XDGData files in fedora.spec file here: https://github.com/FreeCAD/FreeCAD/pull/1741

**Comment by eszlari** [comment]
> I'm not sure if it makes any sense here. It will be always "FreeCAD"?

I just copied it from `package/fedora/freecad.desktop`.

**Comment by yorikvanhavre** [comment]
> @yorikvanhavre FYI, we should add this to a Release TODO list so we don't forget during release time, what do you think ?
That's a good idea. In any case it would be good to have such a release todo list.
> I'm not sure if it makes any sense here. It will be always "FreeCAD"?
It's true, but most other apps do that too (Gimp, Blender, etc...). In any case, the whole xdgdata stuff seems to still generate a lot of discussion on the net as to how to implement it correctly, so let's start as is, and better it over time... Since it is a general spec and should be implemented for all linux distros, it is indeed probably better in a separate directory than inside a distro-specific directory. But inside the debian dir there are also xdg stuff, at some point in time it should all be unified, but that can be done later on.

**Comment by PrzemoF** [comment]
I copied the freecad.desktop file IIRC from the official fedora builds, like here: https://src.fedoraproject.org/cgit/rpms/freecad.git/tree/freecad.desktop :-)
A second thought: non Latin languages might need the translations

**Pull request merged**